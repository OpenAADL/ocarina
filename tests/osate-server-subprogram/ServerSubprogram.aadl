subprogram sub
end sub;

subprogram sub2
end sub2;

data d
  features
    s: subprogram sub2;
end d;

thread t
  features
    ss: server subprogram sub2;
end t;

thread tt
  features
    ss1: server subprogram sub;
    ss2: server subprogram d.s;
    -- bad: t.ss is not a subprogram
    ss3: server subprogram t.ss;
end tt;
