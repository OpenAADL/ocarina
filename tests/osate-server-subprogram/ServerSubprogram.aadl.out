ServerSubprogram.aadl:22:05: ss3 (subprogram spec) points to ss (subprogram spec), which is not of an adequate kind
Cannot analyze AADL specifications
